qcodes_contrib_drivers/drivers/Bilt/ITest.py

```diff
@@ -0,0 +1,475 @@
+        super().__init__(parent, name)
+        
+        self.chan_num = chan_num
+        # Get channel id
+        i, c = round(chan_num/4.)+1, chan_num%4
+        self.chan_id = 'i{};c{};'.format(i,c)
+        
+        self.add_parameter('v',
+                           label='Channel {} voltage'.format(chan_num),
+                           unit='V',
+                           docstring='Voltage of the channel in volt.',
+                           get_cmd=partial(self._parent._get_voltage, chan_num),
+                           get_parser=float,
+                           set_cmd=partial(self._parent._set_voltage, chan_num),
+                           vals=vals.Numbers(-50, 50)
+                           )
+           
+        self.add_parameter('i',
+                           label='Channel {} current'.format(chan_num),
+                           unit='A',
+                           docstring='Current of the channel in ampere.',
+                           get_cmd=partial(self._parent._get_current, chan_num),
+                           get_parser=float
+                           )
+        
+        self.add_parameter('ramp_slope',
+                           label='Channel {} ramp slope'.format(chan_num),
+                           unit='V/ms',
+                           docstring='Slope of the ramp in V/ms.',
+                           get_cmd=partial(self._parent._get_ramp_slope, chan_num),
+                           get_parser=float,
+                           set_cmd=partial(self._parent._set_ramp_slope, chan_num),
+                           vals=vals.Numbers(0, 1)
+                           )
+        
+        self.add_parameter('output_mode',
+                           label='Channel {} output mode'.format(chan_num),
+                           docstring='Mode of the output {exp, ramp}.',
+                           get_cmd=partial(self._parent._get_output_function, chan_num),
+                           get_parser=str,
+                           set_cmd=partial(self._parent._set_output_function, chan_num),
+                           set_parser=str
+                           )
+        
+        self.add_parameter('v_range',
+                           label = 'Channel {} voltage range'.format(chan_num),
+                           docstring='Range of the channel in volt.',
+                           set_cmd=partial(self._parent._set_chan_range, chan_num),
+                           set_parser=float,
+                           get_cmd=partial(self._parent._get_chan_range, chan_num),
+                           get_parser=float)
+        
+        self.add_parameter('state',
+                           docstring='State of the channel {on, off}.',
+                           get_cmd=partial(self._parent._get_chan_state, chan_num),
+                           get_parser=bool,
+                           set_cmd=partial(self._parent._set_chan_state, chan_num),
+                           vals=vals.OnOff())
```

Comment:
as per this section in the docs https://qcodes.github.io/Qcodes/examples/writing_drivers/Creating-Instrument-Drivers.html#On/Off-parameters , i highly recommend to implement this parameter as suggested below and adjust `_get_chan_state`/`_set_chan_state` methods to accept "on"/"off" as input (and not perform the conversion to them from a boolean):
```suggestion
        self.add_parameter('enabled',
                           docstring='Whether the channel is enabled or not',
                           get_cmd=partial(self._parent._get_chan_state, chan_num),
                           set_cmd=partial(self._parent._set_chan_state, chan_num),
                           val_mapping=create_on_off_val_mapping(on_val="on", off_val="off")
```